src/components/ConnectWallet/ConnectWalletDialog.tsx

```diff
@@ -0,0 +1,126 @@
+  height: 183px;
+`
+
+const StyledSmallButton = styled(Button)`
+  background-color: ${({ theme }) => theme.container};
+  border-radius: ${({ theme }) => theme.borderRadius * 0.75}em;
+  height: 90px;
+  padding: 20px;
+`
+
+const StyledNoWalletButton = styled(Button)`
+  background-color: ${({ theme }) => theme.container};
+  border-radius: ${({ theme }) => theme.borderRadius * 0.75}em;
+  height: 90px;
+  padding: 20px;
+`
+
+const StyledNoWalletText = styled(ThemedText.Subhead1)`
+  line-height: 20px;
+  white-space: pre-wrap;
+`
+
+interface ButtonProps {
+  walletName?: string
+  logoSrc?: string
+  caption?: string
+  onClick: () => void
+}
+
+function WalletConnectButton({ walletName, logoSrc, caption, onClick }: ButtonProps) {
+  return (
+    <StyledMainButton onClick={onClick}>
+      <ButtonContents>
+        <img src={logoSrc} alt={walletName} key={walletName} width={100} />
+        <ThemedText.Subhead1>
+          <Trans>{walletName}</Trans>
+        </ThemedText.Subhead1>
+      </ButtonContents>
+    </StyledMainButton>
+  )
+}
+
+function NoWalletButton() {
+  const helpCenterUrl = 'https://help.uniswap.org/en/articles/5391585-how-to-get-a-wallet'
+  return (
+    <StyledNoWalletButton onClick={() => window.open(helpCenterUrl)}>
+      <StyledNoWalletText>
+        <Trans>I don't have a wallet</Trans>
+      </StyledNoWalletText>
+    </StyledNoWalletButton>
+  )
+}
+
+function MetaMaskButton({ walletName, logoSrc, onClick }: ButtonProps) {
+  return (
+    <StyledSmallButton onClick={onClick}>
+      <ButtonContents>
+        <img src={logoSrc} alt={walletName} key={walletName} width={26} />
+        <ThemedText.Subhead1>
+          <Trans>{walletName}</Trans>
+        </ThemedText.Subhead1>
+      </ButtonContents>
+    </StyledSmallButton>
+  )
+}
+
+function MainWalletConnectionOptions({ connection }: { connection: Web3Connection }) {
+  const useWalletConnect = useConnect(connection)
+  return <WalletConnectButton walletName="WalletConnect" logoSrc={WALLETCONNECT_ICON_URL} onClick={useWalletConnect} />
+}
+
+function SecondaryWalletConnectionOptions({ connection }: { connection: Web3Connection }) {
+  const useMetaMask = useConnect(connection)
+  return (
+    <SecondaryOptions>
+      <MetaMaskButton walletName="MetaMask" logoSrc={METAMASK_ICON_URL} onClick={useMetaMask} />
+      <NoWalletButton />
+    </SecondaryOptions>
+  )
+}
+
+export function ConnectWalletDialog() {
+  const [mmConnection, wcConnection] = connections
+  // TODO(kristiehuang): what happens when I try to connect one wallet without disconnecting the other?
+
+  return (
+    <>
+      <Header title={<Trans>Connect wallet</Trans>} />
+      <Body align="stretch" padded open={true}>
+        <Column>
+          <MainWalletConnectionOptions connection={wcConnection} />
+          <SecondaryWalletConnectionOptions connection={mmConnection} />
+        </Column>
+      </Body>
+    </>
+  )
+}
```

Comment:
nit: since these `MainWalletConnectionOptions` and `SecondaryWalletConnectionOptions` are so small and just call a single hook with a prop, I think it'd be a bit easier for future readers if they were just a part of `ConnectWalletDialog` instead. This also removes the last `Main` and `Secondary` usage in favor of the explicit mm/wc content

(not super important though)